### PR TITLE
fix: disable unattended upgrades during CSE execution

### PR DESCRIPTION
**Pull request by sylr**
Signed-off-by: Sylvain Rabot <s.rabot@lectra.com>

**Reason for Change**:

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: parts/k8s/cloud-init/artifacts/cse_install.sh [hunk 1]
```diff
@@ -266,16 +266,24 @@ pullContainerImage() {
...
+        pids="$pids $!"
         docker rmi $(docker images --format '{{.Repository}}:{{.Tag}}' | grep 'nginx') &
+        pids="$pids $!"
     fi
 
     # TODO: remove once ACR is available on Azure Stack
     docker rmi registry:2 &
+    pids="$pids $!"
+
+    wait $pids
```
Comment (jackfrancis) [inline parts/k8s/cloud-init/artifacts/cse_install.sh]: These changes don't specifically relate to disabling unattended upgrades, right? This change would block on returning control from `cleanUpContainerImages` until all the docker images are deleted, right?

Comment (sylr) [reply parts/k8s/cloud-init/artifacts/cse_install.sh]: Yeah I've just realized that.

It relates to the reboot of the node at the end of the cse if security upgrades have been installed.

I'll move it out and make a separate PR.

File: parts/k8s/cloud-init/masternodecustomdata.yml [hunk 2]
```diff
@@ -118,6 +118,26 @@ write_files:
   content: !!binary |
     {{CloudInitData "aptPreferences"}}
 
+- path: /etc/apt/apt.conf.d/10periodic
```
Comment (jackfrancis) [inline parts/k8s/cloud-init/masternodecustomdata.yml]: Do we need this `10periodic` configuration file? I would think that simply removing the `99periodic` configuration with all the `0` values would restore the default unattended upgrade configuration. Are you also suggesting we explicitly re-configure the default values for `Update-Package-Lists`, `Download-Upgradeable-Packages`, `AutocleanInterval`, and `Unattended-Upgrade`?

Comment (sylr) [reply parts/k8s/cloud-init/masternodecustomdata.yml]: I'll come back to you about that cause it's been a while and I don't remember where i've taken the content of `10periodic`

File: parts/k8s/cloud-init/artifacts/cse_install.sh (hunk 1 above)
Comment (jackfrancis) [reply parts/k8s/cloud-init/artifacts/cse_install.sh]: thanks!

File: parts/k8s/cloud-init/masternodecustomdata.yml (hunk 2 above)
Comment (sylr) [reply parts/k8s/cloud-init/masternodecustomdata.yml]: This `10periodic` seems to be part of the `update-notifier-common` package.

I guess we don't need to ship it ourselves.

**Comment by sylr** [comment]
@jackfrancis can we move forward with this ?

**Comment by sylr** [comment]
Guys ?

**Comment by sylr** [comment]
ping

**Pull request merged**